### PR TITLE
DDF-2920 Added initial jenkinsfile for Nightly Linux DDF pipelines build

### PR DESCRIPTION
**Pull request by mcalcote**
#### What does this PR do?
This initial PR is a WIP that's goal is to add the Jenkinsfile to the build and setup the format for improvements for the following tickets.

#### Who is reviewing it (please choose AT LEAST two reviewers that need to approve the PR before it can get merged)?
@clockard @oconnormi @LinkMJB 

**I will add more reviewers/committers after initial feedback or close and reopen when this is in a more finished state**

#### What are the relevant tickets?
[DDF-2920](https://codice.atlassian.net/browse/DDF-2920)

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    environment {
+        // What variables do we need here
+        projectName = 'DDF'
+    }
+    stages {
+        stage('Build') {
+            // Parallel Stage Syntax may soon be changing for declaritive Jenkinsfile(s)
+            // https://issues.jenkins-ci.org/browse/JENKINS-41334
+            steps{
+                parallel (
```
Comment (mcalcote) [inline Jenkinsfile]: pretty sure this isn't the correct usage

File: Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,91 @@
+pipeline {
+    agent {
+        label 'linux-large'
+        //TODO 6146 windows builds ?using label 'proxmox-windows'?
```
Comment (clockard) [inline Jenkinsfile]: might be able to use 'node' syntax so you can have the windows and linux work in parallel.

File: Jenkinsfile [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    }
+    environment {
+        // What variables do we need here
+        projectName = 'DDF'
```
Comment (clockard) [inline Jenkinsfile]: In the future I think this is where we would set variables to indicate if this is a pr build, master build or a nightly build.

File: Jenkinsfile (hunk 1 above)
Comment (clockard) [inline Jenkinsfile]: I think the parallel part might work at a higher level so we could run linux and windows builds in parallel

File: Jenkinsfile [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+                    "linuxBuild" : {
+                        steps {
+                            slackSend color: 'good', message: "STARTED: Job ${BUILD_NUMBER}"
+                            sh """echo Full build of the "$projectName" "$GIT_BRANCH" branch"""
+                            withMaven(maven: 'M3', mavenSettingsConfig: 'codice-maven-settings') {
+                                //TODO update mavenSettingsConfig with a valid config
+                                //sh 'mvn clean install -T 16 -P!documentation -P!findbugs -DskipTests=true -Dpmd.skip=true -Djacoco.skip=true -Dfindbugs.skip=true -Dcheckstyle.skip=true'
+
+                                //run full build here until we can clean the maven phases further
+                                sh 'mvn clean install'
```
Comment (clockard) [inline Jenkinsfile]: I think initially all we want to skip itest here using something like ` -pl \!:test-itests-ddf` and then only build/run the itests in the Integration Test stage.

File: Jenkinsfile [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+                            }
+                        }
+                    }
+                )
+            }
+        }
+        stage('Unit Test') {
```
Comment (clockard) [inline Jenkinsfile]: Until we get our poms/dependencies worked out we can't really run this separately.

File: Jenkinsfile (hunk 2 above)
Comment (mcalcote) [inline Jenkinsfile]: We might need to fallback to the 'node' scripted syntax. I will see how possible it is to manage agents in the declarative pipelines format.

File: Jenkinsfile (hunk 3 above)
Comment (mcalcote) [inline Jenkinsfile]: We can pull a little from the pre-exisiting pipelines environment variables. For the nightly(s) we could use the project names and that will likely be descriptive enough.

File: Jenkinsfile (hunk 5 above)
Comment (mcalcote) [inline Jenkinsfile]: Yeah, I wonder how far off we are....

**Comment by mcalcote** [comment]
Closing to reopen and a full PR

**Pull request closed without merging**